### ISSUE TITLE
Release a new (beta) version

### ISSUE THREAD

**Issue by wdlkmpx** [masneyb/gftp#69]
There have been some changes recently and maybe a new version makes sense, just to let people know that the project is being (slowly) updated.

Some bugs have been fixed and the app is bit more stable, although it still buggy in some parts, bugs that were always there, and probably some new hidden bugs, who knows.

I can produce release tarballs that can be added to github beta releases (red), git tags.

For gtk3 support I see some blockers:

- GtkTreeView part3 - replace GtkCList in Preferences->Local Hosts... I never used this feature and I don't feel like understanding what it does, but reading the code I see the word 'proxy'. It's confusing. But deleting this would make everything simpler I guess.

- GtkTreeView part4 - replace GtkCTree in the bookmarks dialog.. this requires some structural changes in the `gftp_bookmarks_var` struct defined in gftp.h.. I'm not able to understand how to fix this properly, it's just too complex, a project in itself.

- GtkTreeView part5 - replace GtkCTree in the file transfers, I still don't know how hard this is.

- Replace some GDK2 threads stuff that is not compatible with GDK3

After that, all GTK2 deprecations are basically removed. And adding gtk3 support may not be that hard.

Overall, the project is a bit simpler with less obscure code, but the GtkCList and GtkCTree stuff is really obscure, somehow they work fine with the lib/ stuff, but porting the same code to gtk2 is a bit tricky and requires some structural changes or some hacks to make it work properly, but overall I see it's more readable. With the GTK1 stuff there are objects attached to the widget (bookmarks), some GLists and properties somehow just work, and it's hard to tell what is happening. That is not the case with the GTK2 code.

So maybe by releasing a new beta version, there will be more feedback or something. Some gtk gurus may chime in

**Comment by masneyb** [comment]
I think it would be great to release a new beta version of gftp with all of the good work that you've done so far, plus the fixes from others that have accrued over the years on this project. I also think that this would be a good time for you to be the maintainer. I can help review code and answer questions. What do you think about doing the following:

- You create a gftp repo in your github (or gitlab if you prefer). It could be in your own personal space or create a gftp project. Unfortunately github.com/gftp is already taken if you take the latter approach.
- I can setup a HTTP redirect on www.gftp.org that redirects users to the new project. I could link to the github project or the github pages version of the project.
- I can set this gftp project to readonly and update the README pointing to your new project.

What do you think?

**Comment by wdlkmpx** [comment]
Maintaining a web site is time consuming, I see GitHub offers everything a developer wants, including web sites, I created one: https://wdlkmpx.github.io/Xdialog/ (origin: http://xdialog.free.fr/doc/index.html )

There are two possibilities:

- Use this github repo and add collaborators with privileges. I can be one. I see collaborators can be admins as well. Collaborators can edit the wiki pages, create releases (and pre-releases), merge PR's and push directly to the git repo.
https://github.com/masneyb/gftp/access -> Invite collaborator -> wdlkmpx

- Transfer github repo to another user (me) 
https://help.github.com/en/github/administering-a-repository/transferring-a-repository

The target repo is my forked repo (I can delete it). I was thinking of creating an Organization: gee-desktop and add apps, but.. aahh..

Issues, subscribers and everything is kept I think.

I think gftp only needs a GitHub repo (issues, pull requests. releases) .. www.gftp.org could redirect to this git repo or some other url where the repo is transferred.

The git repo needs a cleanup to remove some old stuff, i.e: the ChangeLog is the git log. Remove the /debian directory... remove /docs/website, /docs/figures/mainwin.xcf, etc.

.. to produce smaller tarballs with a script.

**Comment by masneyb** [comment]
I made you a collaborator on this git repo. I don't see an option to make you an admin like I see in the enterprise version of github. I also setup www.gftp.org to forward users to this git repo. So it's completely up to you what to do next.

- We can leave the repo here. If we do that, then feel free to make any changes that you see fit. Feel free to tag me on any PRs that you'd like me to help review.

- I can transfer this repo somewhere else.

Your choice.

**Comment by wdlkmpx** [comment]
I think I can now do everything I want: I can push changes directly, I can merge my own PR's, I can create releases, I can edit the wiki, I guess that's all.

I'll keep opening PR's for review, the release will probably happen on next monday.. I suggest a pre-release, version 2.6b or 2.6.0b, with some notes:

```
Highlights:

- removed GTK1 support
- removed HTTP support (broken)
- removed FSP support
- various bug fixes and improvements
- requires gtk 2.24+ / glib 2.32+
- ongoing work to add GTK3 support
```

**Comment by masneyb** [comment]
That sounds good! There's been a fair number of little changes since December 2008 that haven't been in an official release. Two other things that should be called out is that there's probably a dozen or more language translations that were updated, and at least two new ones based on the git changelog.

It'd be good to mention all of the contributors since the last release in the announcement.

**Comment by wdlkmpx** [comment]
Preparations continue. A new item to release notes:
`- about a dozen language translations were updated`

Contributors, I updated the AUTHORS file
commit: https://github.com/masneyb/gftp/commit/662c4eb472393c61a6016c435849f24c57917982

The AUTHORS file is installed to ${docdir} (/usr/share/doc/gftp)

It now includes all the people who contributed to the CVS + the THANKS file.

I used this command: `# git log --all --format='%cN' | sort -u `
and removed a couple invalid entries

I didn't include emails, as search engines may index them, some people might not want that.

However it's easy to include emails with this command:
`# git log --all --format='%cN <%cE>' | sort -u`

**Comment by wdlkmpx** [comment]
PO files. I see updating these files is a big task. It's easy to update them all with a script, but it's hard to keep some translations.

Some po files haven't been updated in 15 years or more and some translations no longer apply. 
be.po:
```
#: lib/config_file.c:724
msgid ""
"Config file for gFTP. Copyright (C) 1998-2003 ....
```
This probably got broken around 2007-2008.

There are new strings. Some files were heavily edited and that may cause some problems. Some files were deleted. Some strings were intentionally deleted. See https://github.com/masneyb/gftp/commit/38bcda9378a19cb21fc9fad4ac65285ff605d70b

From now on I'll keep all .po files up to date, that way people only need to edit .po files and open PR's or attach them to a comment.

But I guess all the changes I want to make should happen as soon as posible, because a single update is a big commit. Some big changes happened.

The way I see it, it will be easy to update translations, I'll create a wiki page or something with simple instructions.

**Comment by wdlkmpx** [comment]
I see the algorithm that updates .po files is somewhat intelligent. In the update process it's able to identify patterns or something

uk.po (Ukrainian)

```
#: ../src/gtk/gftp-gtk.c:472
#, fuzzy
msgid "_Remote"
msgstr "/_Віддалено"
```
````
#: ../src/gtk/gftp-gtk.c:521
#, fuzzy
msgid "Window _1"
msgstr "/FTP/Вікно 1"
```

Window _1 = GtkUIManager (updated string from gftp-gtk.c)
/FTP/Вікно 1 = GtkItemFactory

So the correct translation is:

```
#: ../src/gtk/gftp-gtk.c:472
msgid "_Remote"
msgstr "_Віддалено"
```
````
#: ../src/gtk/gftp-gtk.c:521
msgid "Window _1"
msgstr "Вікно _1"
```

I figured out that updating .po files is the last thing I should do before a release. And all .po files need to be fixed manually, this can easily take 2 weeks or a month.

There are some files that I can't edit or fix, i.e: ar.po (Arabic). Hmmm..

**Comment by masneyb** [comment]
Generally I would just leave the PO files alone and let the translators update the files. Back when I had this in GNOME's git, the translators would directly commit the updates into the repo.

**Comment by wdlkmpx** [comment]
I see the translation of the menus is broken. When a string changes, the translation no longer applies. `GtkItemFactory -> GtkUIManager`

I generated a new gftp.pot, which can be used as a reference or template.

Most of the work can be done with `make -C po update-po`, but it requires native speakers to fix issues that arise. I guess I can fully update 1 po file, and fix a few other po files with similar languages.

**Comment by masneyb** [comment]
I would just update the pot file and let the translators touch the po files. If a translation is not available, then it'll fall back to English. You'll get PRs to address these from the community.

Two things to keep in mind to help the translators:

- Try to avoid unnecessarily changing strings and try to reuse them if it makes sense.
- Avoid joining strings like this pseudo code: `"The file is named " + name + "  and has size " + size`. Do this instead: `"The file is named %s and has size %lld", name, size`. For cases where the arguments need to be swapped for a translation, the translator can do that in their string with `%1$s` and `%2$lld` to reference a specific argument.

**Comment by wdlkmpx** [comment]
I created a git tag https://github.com/masneyb/gftp/releases/tag/2.0.19

And I'm almost ready to create a new release (pre-release - red). 

Github automatically creates a tarball for each release, but that's not the traditional release tarball, I'll create it with the `zrelease` script and add it as part of the release.

But one thing is fore sure, the autotools stuff creates too many files, it pollutes the source tree a bit. There must be a way to move all (or most of) those files to a subdirectory.

The ChangeLog will contain the highlights + `git shortlog 2.0.19..HEAD`.

**Comment by wdlkmpx** [comment]
I created a new release: 2.6.0b.. I think it should have been 2.6.0beta, but it's already done. It's not a pre-release because pre-releases are not shown in the main page. 

It's more stable than 2.0.19, it compiles in very recent distros, some bugs were fixed, other bugs were deleted. And in my tests the new code works fine.

I'll keep this issue open for a while, some important info has been revealed here.

Now I think it's time to add some topics, but I can't do that:
https://help.github.com/en/github/administering-a-repository/classifying-your-repository-with-topics

Topics:
`ftp ftps sftp ssh2 gtk gtk2 gtk3 client`
